src/include/catalog/database_catalog.h

```diff
@@ -238,6 +238,10 @@ class DatabaseCatalog {
   std::vector<std::pair<common::ManagedPointer<storage::index::Index>, const IndexSchema &>> GetIndexes(
       transaction::TransactionContext *txn, table_oid_t table);
 
+  bool InsertLanguage(transaction::TransactionContext *txn, const std::string &lanname);
+
+  bool DeleteLanguage(transaction::TransactionContext *txn, const std::string &lanname);
```

Comment:
To be consistent with the rest of the catalog, you should only support deletions by OID.